src/storage/src/compaction/picker.rs

```diff
@@ -112,7 +127,7 @@ impl<S: LogStore> Picker for SimplePicker<S> {
 
         for level_num in 0..levels.level_num() {
             let level = levels.level(level_num as u8);
-            let outputs = self.strategy.pick(ctx, level);
+            let (compaction_time_window, outputs) = self.strategy.pick(ctx, level);
```

Comment:
Then you can build a `PickerContext` with `compaction_time_window` here, and pass the context to `self.strategy.pick` here to let compaction strategy know the original `compaction_time_window` value of that table.